New Sheet - Shadowdark Unofficial character sheet for PCs and NPCs

nothingisnecessary opened the pull request:
# Submission Checklist
## Required

<!-- Check these off by adding an 'x' to each of these boxes. If you fail to meet all these criteria, your PR will be rejected. -->

- [x] The pull request title clearly contains the name of the sheet I am editing.
- [x] The pull request title clearly states the type of change I am submitting (New Sheet/New Feature/Bugfix/etc.).
- [x] The pull request makes changes to files in only one sub-folder.
- [x] The pull request does not contain changes to any json files in the translations folder (translation.json is permitted)

## New Sheet Details

- The name of this game is: Shadowdark
- The publisher of this game is: The Arcane Library LLC
- The name of this game system/family is: Shadowdark / OSR

- [x] I have followed the [Character Sheets Standards](https://wiki.roll20.net/Building_Character_Sheets#Roll20_Character_Sheets_Repository) when building this sheet.

<!-- Please check any that apply: -->

- [ ] This sheet has been made on behalf of, or by, the game's publisher.
- [x] This game is not a traditionally published game, but a copy of the game rules can be purchased/downloaded/found at: https://www.thearcanelibrary.com/pages/shadowdark
- [ ] This sheet is for an unofficial fan game, modification to an existing game, or a homebrew system.

# Changes / Description

- Created a sheet for use with Shadowdark RPG by The Arcane Library LLC
- Included third-party license text to be compliant with rules set forth by The Arcane Library LLC in the Shadowdark RPG core rules
- Tested OK in Chrome and Firefox, thanks to the members of The Arcane Library discord server for helping test this!
- Includes actions for importing sheets from JSON (Shadowdarklings.net format) and for importing monsters by parsing text from the book, but doesn't include any license-infringing content (such as level progression traits or character creation)
- Includes level-based XP limits
- Includes Luck Token tracking and an optional rule for "Pulp Mode" which enables tracking 6 luck tokens instead of the usual 1 luck token
- The inventory slots capacity must be updated by hand, but doing so will automatically 'disable' unusable slots by graying them out (no data is lost; I avoided automatic update of max slots because, while it was easy to do for most classes based on  `Math.max(Strength, 10)` it can optionally depend on a class feature which may have been borderline license violation)
- Intentionally doesn't look like the official sheet, but is more functional like 5e sheets and the HTML was built from scratch to work well with roll20 (The official sheet artwork and Logo aren't OK to use without permission from The Arcane Library)
- While the HTML was built from scratch, I borrowed and edited some roll templates and JavaScript from the 5e Darker Dungeons sheet (thanks GiffyGlyp!), especially the code for updating repeating attacks based on stats.
- Responsive flex design which can also be used on smartphones (as much as roll20 can do this; I tested it on Pixel 8)
- CSS for Darkmode and standard "light" mode support (`.sheet-darkmode`)

Thank you for your consideration
- CyberSasquatch (thescratch)

roll20deploy commented:
[Character Sheet Info](https://app.roll20.net/managesheets/Shadowdark-Unofficial/roll20-character-sheets) *Roll20 Internal Use only.*

nothingisnecessary commented:
Fixed bugs, UI, and updated preview image

nothingisnecessary commented:
Updated credits and removed unused option. Sheet functions and roll templates tested and working OK for NPC and PC modes. Styles look OK for normal and dark modes.
Ready for review and merge!
